File: Backend/LogisticsPlatform.Application/DTOs/Customers/CreateCustomerFullDto.cs
using LogisticsPlatform.Application.DTOs.Customers.Contacts;
using LogisticsPlatform.Application.DTOs.Customers.Notes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticsPlatform.Application.DTOs.Customers
{
    public class CreateCustomerFullDto
    {
        public CreateCustomerDto Customer { get; set; } = null!;
        public List<CreateCustomerAddressDto> Addresses { get; set; } = new();
        public List<CreateCustomerContactDto> Contacts { get; set; } = new();
        public List<CreateCustomerNoteDto> Notes { get; set; } = new();
    }

}

File: Frontend/src/app/core/models/customers/create-customer-full.dto.ts
import { CreateCustomerDto } from './create-customer.dto';
import { CreateCustomerAddressDto } from './addresses/create-customer-address.dto';
import { CreateCustomerContactDto } from './contacts/create-customer-contact.dto';
import { CreateCustomerNoteDto } from './notes/create-customer-note.dto';

export interface CreateCustomerFullDto {
    customer: CreateCustomerDto;
    addresses: CreateCustomerAddressDto[];
    contacts: CreateCustomerContactDto[];
    notes: CreateCustomerNoteDto[];
}
